### PR TITLE
fix: error handling for trigger release

### PR DESCRIPTION
**Pull request by Ash-exp**
# Description
With `ENABLE_ASYNC_INSTALL_DEVTRON_CHART` flag enabled, the error statements are not getting propagated in trigger deployment

Fixes https://github.com/devtron-labs/devtron/issues/4490

## How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

## Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR requires documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code.
* [x] I have commented my code, particularly in hard-to-understand areas.
* [ ] I have tested it for all user roles.
* [ ] I have added all the required unit/api test cases.

## Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

File: pkg/pipeline/WorkflowDagExecutor.go [hunk 1]
```diff
@@ -2080,9 +2080,18 @@ func (impl *WorkflowDagExecutorImpl) TriggerDeployment(cdWf *pipelineConfig.CdWo
...
 
 	releaseErr := impl.TriggerCD(artifact, cdWf.Id, savedWfr.Id, pipeline, triggeredAt)
+	// if releaseErr found, then the mark current deployment Failed and return
+	if releaseErr != nil {
+		err := impl.MarkCurrentDeploymentFailed(runner, releaseErr, triggeredBy)
+		if err != nil {
+			impl.logger.Errorw("error while updating current runner status to failed, updatePreviousDeploymentStatus", "cdWfr", runner.Id, "err", err)
+			return releaseErr
+		}
+		return releaseErr
```
Comment (kripanshdevtron) [inline pkg/pipeline/WorkflowDagExecutor.go:2090]: if we are returning releaseErr in both cases, then we should make it common.

File: pkg/pipeline/WorkflowDagExecutor.go [hunk 2]
```diff
@@ -2422,12 +2422,20 @@ func (impl *WorkflowDagExecutorImpl) ManualCdTrigger(overrideRequest *bean.Value
...
 		span.End()
-
+		// if releaseErr found, then the mark current deployment Failed and return
+		if releaseErr != nil {
+			err := impl.MarkCurrentDeploymentFailed(runner, releaseErr, overrideRequest.UserId)
+			if err != nil {
+				impl.logger.Errorw("error while updating current runner status to failed, updatePreviousDeploymentStatus", "cdWfr", runner.Id, "err", err)
+				return 0, releaseErr
+			}
+			return 0, releaseErr
```
Comment (kripanshdevtron) [inline pkg/pipeline/WorkflowDagExecutor.go:2432]: make return stmt common

File: pkg/pipeline/WorkflowDagExecutor.go (hunk 1 above)
Comment (Ash-exp) [reply pkg/pipeline/WorkflowDagExecutor.go]: updated 👍

File: pkg/pipeline/WorkflowDagExecutor.go (hunk 2 above)
Comment (Ash-exp) [reply pkg/pipeline/WorkflowDagExecutor.go]: updated 👍

**Review by kripanshdevtron — APPROVED** [review]

**Pull request merged**